pkg/services/accesscontrol/authorizer.go

```diff
@@ -82,43 +83,42 @@ type LegacyAccessClient struct {
 	opts map[string]ResourceAuthorizerOptions
 }
 
-// HasAccess implements claims.AccessClient.
-func (c *LegacyAccessClient) HasAccess(ctx context.Context, id claims.AuthInfo, req claims.AccessRequest) (bool, error) {
+func (c *LegacyAccessClient) Check(ctx context.Context, id claims.AuthInfo, req authz.CheckRequest) (authz.CheckResponse, error) {
 	ident, ok := id.(identity.Requester)
 	if !ok {
-		return false, errors.New("expected identity.Requester for legacy access control")
+		return authz.CheckResponse{}, errors.New("expected identity.Requester for legacy access control")
 	}
 
 	opts, ok := c.opts[req.Resource]
 	if !ok {
 		// For now we fallback to grafana admin if no options are found for resource.
 		if ident.GetIsGrafanaAdmin() {
-			return true, nil
+			return authz.CheckResponse{Allowed: true}, nil
```

Comment:
Self note: We should expose `CheckResponseAllowed` and `CheckResponseDenied` from the library.

Reply:
No I don't think that is necessary